controllers/component_controller.go

```diff
@@ -382,7 +378,7 @@ func (r *ComponentReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 
 	// Get the Webhook from the event listener route and update it
 	// Only attempt to get it if the build generation succeeded, otherwise the route won't exist
-	if component.Status.Conditions[len(component.Status.Conditions)-1].Status == v1.ConditionTrue && component.Spec.Source.ImageSource == nil {
+	if component.Status.Conditions[len(component.Status.Conditions)-1].Status == v1.ConditionTrue && component.Spec.ContainerImage != "" {
```

Comment:
```suggestion
	if component.Status.Conditions[len(component.Status.Conditions)-1].Status == v1.ConditionTrue && component.Spec.Source.GitSource != nil && component.Spec.Source.GitSource.URL != "" {
```

We don't want to get the webhook for container image components. Only when there is a git source component.

Reply:
updated